src/main/scala/nodes/learning/LeastSquaresEstimator.scala

```diff
@@ -0,0 +1,87 @@
+package nodes.learning
+
+import breeze.linalg._
+import nodes.util.{Densify, Sparsify}
+import org.apache.spark.rdd.RDD
+import pipelines.Logging
+import workflow.{Pipeline, WeightedNode, LabelEstimator, OptimizableLabelEstimator}
+
+import scala.reflect._
+
+/**
+ * A least squares solver that is optimized to use a fast algorithm, based on characteristics of the
+ * workload and cost models.
+ *
+ * Currently selects between Dense LBFGS, Sparse LBFGS, Exact Distributed Solve, and Approximate Block Solve.
+ *
+ * The default weights were determined empirically via results run on a 16 r3.4xlarge node cluster.
```

Comment:
So this is a bit unrelated to this PR but it would be good to check in a weights estimator script in `scripts/` or something like that.